devices/ikea.js

```diff
@@ -147,21 +147,24 @@ const ikea = {
                 if (key == 'fan_state' && value.toLowerCase() == 'on') {
                     value = getMetaValue(entity, meta.mapped, 'fanStateOn', 'allEqual', 'on');
                 }
+                else {
+                    value = value.toString().toLowerCase();
+                }
 
                 let fanMode;
-                switch (value.toLowerCase()) {
+                switch (value) {
                 case 'off':
                     fanMode = 0;
                     break;
                 case 'auto':
                     fanMode = 1;
                     break;
                 default:
-                    fanMode = parseInt(((parseInt(value) / 2.0) * 10) + 5);
+                    fanMode = (+value / 0.2) + 5;
```

Comment:
The +value here is very confusing to read, can this go back to parseInt?

Reply:
Sure, reverted the unary plus operator. For me it got an error when setting fan mode from HA.